module-apps/application-settings/windows/advanced/CPUModeTestWindow.cpp

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) 2017-2021, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include "CPUModeTestWindow.hpp"
+
+#include <system/messages/SentinelRegistrationMessage.hpp>
+#include <application-settings/windows/WindowNames.hpp>
+#include <Application.hpp>
+#include <TextFixedSize.hpp>
+#include <BoxLayout.hpp>
+
+#include <magic_enum.hpp>
+
+namespace magic_enum
+{
+    template <> struct enum_range<bsp::CpuFrequencyHz>
+    {
+        static constexpr int min = 0;
+        static constexpr int max = 528;
+    };
+} // namespace magic_enum
+
+namespace gui
+{
+    CPUModeTestWindow::CPUModeTestWindow(app::ApplicationCommon *app) : AppWindow(app, window::name::cpu_test_window)
```

Comment:
That is only Dev Advanced window -> it has no localization and proper style variables as it is not for clients. It is not attached in Release builds.